HaXml uses Data.ByteString.Lazy.Char8.join which exists in
bytestring-0.9.0.1 distributed with ghc-6.8.2, but not any other 0.9.x
version (including the bytestring-0.9.0.1 distributed on hackage).

This patch makes it use intercalate instead, which is the exact same thing
as join, and exists in at least all 0.9.0.x and the (currently) most recent
0.9.1.0.

--

diff -rN -u old-HaXml-1.19.2/src/Text/XML/HaXml/ByteStringPP.hs new-HaXml-1.19.2/src/Text/XML/HaXml/ByteStringPP.hs
--- old-HaXml-1.19.2/src/Text/XML/HaXml/ByteStringPP.hs	2008-04-28 22:58:24.000000000 +0200
+++ new-HaXml-1.19.2/src/Text/XML/HaXml/ByteStringPP.hs	2008-04-28 22:58:24.000000000 +0200
@@ -24,7 +24,7 @@
 import Maybe hiding (maybe)
 import Data.List (intersperse) 
 --import Data.ByteString.Lazy hiding (pack,map,head,any,singleton,intersperse,join)
-import Data.ByteString.Lazy.Char8 (ByteString(), concat, pack, singleton, join
+import Data.ByteString.Lazy.Char8 (ByteString(), concat, pack, singleton, intercalate
                                   , append, elem, empty)
 import Text.XML.HaXml.Types
 
@@ -59,8 +59,8 @@
 ($+$)        = ($$)
 
 hcat = Data.ByteString.Lazy.Char8.concat 
-hsep = Data.ByteString.Lazy.Char8.join (singleton ' ')
-vcat = Data.ByteString.Lazy.Char8.join (singleton '\n')
+hsep = Data.ByteString.Lazy.Char8.intercalate (singleton ' ')
+vcat = Data.ByteString.Lazy.Char8.intercalate (singleton '\n')
 cat  = hcat
 sep  = hsep
 text = pack

